tests/model_explainability/guardrails/conftest.py

```diff
@@ -273,111 +312,169 @@ def huggingface_sr(
-        name=ORCHESTRATOR_CONFIGMAP_NAME,
+        name=QWEN_ISVC_NAME,
         namespace=model_namespace.name,
-        data={
-            "config.yaml": yaml.dump({
-                "chat_generation": {
-                    "service": {
-                        "hostname": f"{QWEN_ISVC_NAME}-predictor.{model_namespace.name}.svc.cluster.local",
-                        "port": GUARDRAILS_ORCHESTRATOR_PORT,
-                    }
-                },
-                "detectors": {
-                    "regex": {
-                        "type": "text_contents",
-                        "service": {
-                            "hostname": "127.0.0.1",
-                            "port": Ports.REST_PORT,
-                        },
-                        "chunker_id": "whole_doc_chunker",
-                        "default_threshold": 0.5,
-                    }
-                },
-            })
+        deployment_mode=KServeDeploymentType.RAW_DEPLOYMENT,
+        model_format="vLLM",
+        runtime=vllm_runtime.name,
+        storage_key=minio_data_connection.name,
+        storage_path="Qwen2.5-0.5B-Instruct",
+        wait_for_predictor_pods=False,
+        resources={
+            "requests": {"cpu": "1", "memory": "8Gi"},
+            "limits": {"cpu": "2", "memory": "10Gi"},
         },
-    ) as cm:
-        yield cm
+    ) as isvc:
+        yield isvc
 
 
 @pytest.fixture(scope="class")
-def gorch_with_hf_detectors_configmap(
+def prompt_injection_detector_isvc(
     admin_client: DynamicClient,
     model_namespace: Namespace,
-) -> Generator[ConfigMap, Any, Any]:
-    with ConfigMap(
+    minio_data_connection: Secret,
+    huggingface_sr: ServingRuntime,
+) -> Generator[InferenceService, Any, Any]:
+    with create_isvc(
         client=admin_client,
-        name=ORCHESTRATOR_CONFIGMAP_NAME,
+        name="prompt-injection-detector",
         namespace=model_namespace.name,
-        data={
-            "config.yaml": yaml.dump({
-                "chat_generation": {
-                    "service": {
-                        "hostname": f"{QWEN_ISVC_NAME}-predictor",
-                        "port": GUARDRAILS_ORCHESTRATOR_PORT,
-                    }
-                },
-                "detectors": {
-                    "prompt_injection": {
-                        "type": "text_contents",
-                        "service": {
-                            "hostname": "prompt-injection-detector-predictor",
-                            "port": 8000,
-                        },
-                        "chunker_id": "whole_doc_chunker",
-                        "default_threshold": 0.5,
-                    }
-                },
-            })
+        deployment_mode=KServeDeploymentType.RAW_DEPLOYMENT,
+        model_format="guardrails-detector-huggingface",
+        runtime=huggingface_sr.name,
+        storage_key=minio_data_connection.name,
+        storage_path="deberta-v3-base-prompt-injection-v2",
```

Comment:
Same here, maybe it is not good to hard code model name to the path. Path can probably be more generic.